### PR TITLE
make php-xml mandatory- add utils for db and admin user creation - install spamassassin from repo

### PR DESCRIPTION
**Pull request by Skywalker-11**
solves #35

### CODE REVIEW

**Comment by stefaweb** [comment]
Great. Will try later.

**Comment by Skywalker-11** [comment]
Now it will always install the most current release of MailWatch and MailScanner

**Comment by stefaweb** [comment]
Works!

All binaries install problems seems solved.

After have configured apache, mailscanner and activated spamassassin (postfix still unconfigured), all seems ok in the GUI.

Except:

clamAV is not see in the GUI
geoIP update do "nothing" in GUI.

sc

MailScanner Lint (Test)
sc2

**Comment by stefaweb** [comment]
EDIT

And this in the root@localhost mailbox (tens of this):

`antispam-host : Jul 15 16:50:47 : www-data : user NOT in sudoers ; TTY=unknown ; PWD=/opt/mailwatch/public ; USER=root ; COMMAND=/usr/sbin/MailScanner --lint`

tools/sudo/mailwatch sudo file is absent from /etc/sudoers.d/ after install.

File: setup.scripts/mailwatch/mailwatch.inc [hunk 1]
```diff
@@ -274,9 +274,9 @@ function install-sudo() {
     else
         logprint "Not installing sudo."
     fi
-    if ! ( type "sudo" > /dev/null 2>&1 ) ; then
+    if ( type "sudo" > /dev/null 2>&1 ) ; then
         logprint "Sudo is installed so we are configuring it for use with MailWatch"
-        sed -i -e "s/User_Alias MAILSCANNER = .*/User_Alias MAILSCANNER = $Webuser" "$MailWatchTmpDir/tools/sudo/mailwatch"
+        sed -i -e "s/User_Alias MAILSCANNER = .*/User_Alias MAILSCANNER = $Webuser/" "$MailWatchTmpDir/tools/sudo/mailwatch"
```
Comment (stefaweb) [inline setup.scripts/mailwatch/mailwatch.inc:279]: Did you change the setup depending of the MTA?

```
# If you use Exim enable these
#Cmnd_Alias EXIM_QUEUE_IN = /usr/sbin/exim -bpc
#Cmnd_Alias EXIM_QUEUE_OUT = /usr/sbin/exim -bpc -DOUTGOING
#Cmnd_Alias SENDMAIL_QUEUE_IN = /usr/bin/mailq -bp -OQueueDirectory=/var/spool/mqueue.in
#Cmnd_Alias SENDMAIL_QUEUE_OUT = /usr/bin/mailq -bp
#MAILSCANNER ALL= NOPASSWD: MAILSCANLINT, SPAMASSASSINLINT, EXIM_QUEUE_IN, EXIM_QUEUE_OUT, SENDMAIL_QUEUE_IN, SENDMAIL_QUEUE_OUT

# If you use postfix enable these
#Cmnd_Alias POSTFIX_QUEUE = /usr/sbin/postqueue
#MAILSCANNER ALL= NOPASSWD: MAILSCANLINT, SPAMASSASSINLINT, POSTFIX_QUEUE

```

Comment (Skywalker-11) [reply setup.scripts/mailwatch/mailwatch.inc:279]: Not yet

File: setup.scripts/apache/mailwatch-apache.inc [hunk 2]
```diff
@@ -24,26 +24,34 @@ function install-webserver {
...
+    logprint "Using apache conf file $mwApacheConf"
+
     #backup old config
     if [ -f "$mwApacheConf" ]; then
         mv "$mwApacheConf" "$mwApacheConf.old"
     fi
 
 #    cp "$InstallFilesFolder/setup.scripts/apache/etc/apache2/conf-available/mailwatch.conf" /etc/apache2/conf-available/mailwatch.conf
-    cp "$InstallFilesFolder/setup.scripts/apache/etc/apache2/sites-available/mailwatch.conf" "$mwApacheConf"
+    cp "$InstallFilesFolder/setup.scripts/apache/etc/apache2/sites-available/ssl_mailwatch.conf" "$mwApacheConf"
```
Comment (stefaweb) [inline setup.scripts/apache/mailwatch-apache.inc:43]: If we follow the "Debian rule", `ssl_mailwatch.conf` will be better as `mailwatch-ssl.conf`.

Comment (Skywalker-11) [reply setup.scripts/apache/mailwatch-apache.inc:43]: At least for Centos it has to load after the ssl module (/etc/httpd/conf.d/ssl.conf)
But I guess we can keep the mailwatch-ssl.conf for debian https://github.com/mailwatch/mailwatch-install-script/pull/38/files/cbfef4fd0974269df318e3fe936e96df4949db4d..7b5765ec584c048801fafb3b0a68d40f0d9f9f13#diff-196ada4b2f7f4a03baef817c4fada680R32

Comment (Skywalker-11) [reply setup.scripts/apache/mailwatch-apache.inc:43]: done

File: setup.scripts/spamassassin/mailwatch-spamassassin.inc [hunk 3]
```diff
@@ -10,14 +10,28 @@ function prepare-spamassassin-install() {
...
+    logprint ""
     if [ "$installSpamassassin" == "y" ]; then
         logprint "Start install of spamassassin"
         $PM install spamassassin
     else
         logprint "Not installing spamassassin"
     fi
+    if [ "$installBayesSql" == "y" ]; then
+        logprint "Configure Bayes-Sql"
+        #mysql -u "$SqlRoot" -p"$SqlRootPwd" -h "$SqlHost" < "$InstallFilesFolder/setup.scripts/spamassassin/bayesSql.sql"
```
Comment (stefaweb) [inline setup.scripts/spamassassin/mailwatch-spamassassin.inc]:
```
# For MySQL/MariaDB
#mysql -u "$SqlRoot" -p"$SqlRootPwd" -h "$SqlHost" < "/usr/share/doc/spamassassin/sql/bayes_mysql.sql"
#mysql -u "$SqlRoot" -p"$SqlRootPwd" -h "$SqlHost" < "/usr/share/doc/spamassassin/sql/awl_mysql.sql"

# For PostgreSQL
#mysql -u "$SqlRoot" -p"$SqlRootPwd" -h "$SqlHost" < "/usr/share/doc/spamassassin/sql/bayes_pg.sql"
#mysql -u "$SqlRoot" -p"$SqlRootPwd" -h "$SqlHost" < "/usr/share/doc/spamassassin/sql/awl_pg.sql"
```

Comment (stefaweb) [reply setup.scripts/spamassassin/mailwatch-spamassassin.inc]: Do you have a function to create user/password and database in setup.sh?

Comment (Skywalker-11) [reply setup.scripts/spamassassin/mailwatch-spamassassin.inc]: They are read from user input
https://github.com/Skywalker-11/mailwatch-install-script/blob/132d2f8762188d86ea351e193401286a967eee79/setup.scripts/mailwatch/mailwatch.inc#L186-L214

The DB is generated by the create.sql from mailwatch repo.

And now that I looked at it I noticed that the database is always `mailscanner` and user input for db is ignored 😞

Comment (Skywalker-11) [reply setup.scripts/spamassassin/mailwatch-spamassassin.inc]: fixed

File: setup.scripts/mailwatch/mailwatch.inc (hunk 1 above)
Comment (Skywalker-11) [reply setup.scripts/mailwatch/mailwatch.inc:279]: fixed in https://github.com/mailwatch/mailwatch-install-script/pull/38/commits/59e0d3f8f31f75a75d960094c7965db8a6593bbd

**Pull request closed without merging**